src/environment.cpp

```diff
@@ -881,13 +885,11 @@ class ABMHandler
 			content_t c = n.getContent();
 			v3s16 p = p0 + block->getPosRelative();
 
-			std::map<content_t, std::vector<ActiveABM> >::iterator j;
-			j = m_aabms.find(c);
-			if(j == m_aabms.end())
+			if (!m_aabms[c])
```

Comment:
For some reason, lhofhansl's comment does not seem to be showing up here so I'll reiterate it:

This needs to check if c >= m_aabms.size() first, otherwise it will access memory out-of-bounds.